### ISSUE TITLE
IPSec tunnel opened/connected but no traffic | If route added manually it works perfect [Site-to-Site]

### ISSUE THREAD

**Issue by Bubelbub** [xelerance/Openswan#225]
Hey guys,

I dont know why it is not working.
Some people told to me that I not need a route in linux routing table for successful ping to each computer in other network.
But if I add the route manually it works perfect.

My network is simple.
I have a IPSec with OpenSwan (IPCop) on the other side and another IPSec with OpenSwan (IPCop) on the other side.
One side is my Server which interface is directly assigned to the public internet address.
The other side is my Client which is connected to a LTE/UMTS stick and uses the internal ip address.

I call them simply "Server" and "Client" - because the Server shouldnt connect to the client but the client should connect to the server automatically.

**Server:**

`ipsec.secrets`

	: RSA /var/ipcop/certs/hostkey.pem
	@srv.dyndns.org @stick.dyndns.org : PSK 'thisismykey'

`ipsec.conf`

	# Do not modify 'ipsec.conf' directly since any changes you make will be
	# overwritten whenever you change IPsec settings using the web interface!
	#

	version 2.0

	config setup
	   protostack=netkey
	   klipsdebug="none"
	   plutodebug="none"
	   uniqueids=yes
	   nat_traversal=yes

	conn %default
	   keyingtries=0
	   disablearrivalcheck=no
	   leftupdown=/usr/local/bin/ipsecupdown.sh

	# 
	# net-2-net to RED
	conn Vereinsnetz
	   left=srv.dyndns.org
	   leftsubnet=192.168.2.0/255.255.255.0
	   right=stick.dyndns.org
	   rightsubnet=192.168.1.0/255.255.255.0
	   leftid="@srv.dyndns.org"
	   rightid="@stick.dyndns.org"
	   ike=aes128-sha-modp1536,aes128-sha-modp1024,aes128-md5-modp1536,aes128-md5-modp1024,3des-sha-modp1536,3des-sha-modp1024,3des-md5-modp1536,3des-md5-modp1024
	   esp=aes128-sha1,aes128-md5,3des-sha1,3des-md5
	   ikelifetime=1h
	   keylife=8h
	   dpddelay=30
	   dpdtimeout=120
	   dpdaction=hold
	   pfs=yes
	   authby=secret
	   auto=add

**Client:**

`ipsec.secrets`

	: RSA /var/ipcop/certs/hostkey.pem
	@srv.dyndns.org @stick.dyndns.org : PSK 'thisismykey'

`ipsec.conf`

	# Do not modify 'ipsec.conf' directly since any changes you make will be
	# overwritten whenever you change IPsec settings using the web interface!
	#

	version 2.0

	config setup
	   protostack=netkey
	   klipsdebug="none"
	   plutodebug="none"
	   uniqueids=yes
	   nat_traversal=yes

	conn %default
	   keyingtries=0
	   disablearrivalcheck=no
	   leftupdown=/usr/local/bin/ipsecupdown.sh

	# 
	# net-2-net to RED
	conn Vereinsnetz
	   left=192.168.1.1
	   leftsubnet=192.168.1.0/255.255.255.0
	   right=srv.dyndns.org
	   rightsubnet=192.168.2.0/255.255.255.0
	   leftid="@stick.dyndns.org"
	   rightid="@srv.dyndns.org"
	   ike=aes128-sha-modp1536,aes128-sha-modp1024,aes128-md5-modp1536,aes128-md5-modp1024,3des-sha-modp1536,3des-sha-modp1024,3des-md5-modp1536,3des-md5-modp1024
	   esp=aes128-sha1,aes128-md5,3des-sha1,3des-md5
	   ikelifetime=1h
	   keylife=8h
	   dpddelay=30
	   dpdtimeout=120
	   dpdaction=restart
	   pfs=yes
	   authby=secret
	   auto=start

Thats working fine. Tunnel is up and running - I would say 👍 

But if I try...

`$ ping 192.168.2.1` or `$ ping 192.168.1.1` (from Client -> Server or Server -> Client) I get no response.
TCPDump shows nothing.
The ping said **No route to host**.
If I try `traceroute` then they go a wrong way. (Use internet gateway instead using IPSec tunnel)

**$ route** (of Server)

	Kernel IP routing table
	Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
	default         static.225.20.7 0.0.0.0         UG    0      0        0 wan-1
	10.45.158.0     10.45.158.2     255.255.255.0   UG    0      0        0 tun0
	10.45.158.2     *               255.255.255.255 UH    0      0        0 tun0
	144.73.20.24   *               255.255.255.248 U     0      0        0 wan-1
	192.168.2.0      *               255.255.255.0   U     0      0        0 lan-1

The 192.168.1.0 network of the client is missing.

And... If I'm adding the routes manually.

**Server:** `route add -net 192.168.1.0/24 gw 192.168.2.1 lan-1`
**Client:** `route add -net 192.168.2.0/24 gw 192.168.1.1 lan-1`

Then it work's perfect.

	traceroute to 192.168.1.1 (192.168.1.1), 30 hops max, 60 byte packets
	 1  router.local (192.168.2.1)  0.172 ms  0.227 ms  0.176 ms
	 2  192.168.1.1 (192.168.1.1)  26.148 ms  26.398 ms  26.453 ms

But Why!?!? the route is not added automatically from IPSec?
On `auto` it should `add` on server and `start` on client.
So why IPSec / OpenSwan not add this routes automatically?

And in my first sentence I said that somebody said IPSec should work without routes.
So why it work if I add the route and not if I not add the route!? 👎 

**Some more informations:**

**$ ip xfrm state**

	src 87.172.164.149 dst 144.73.20.30
			proto esp spi 0x1056faee reqid 16405 mode tunnel
			replay-window 32 flag af-unspec
			auth-trunc hmac(sha1) 0x9310c236f4917891fe9a5b3215cd2b3a801d593c 96
			enc cbc(aes) 0xca1767ae12dd3cdf85cbbdbcedce2c6c
			encap type espinudp sport 63453 dport 4500 addr 0.0.0.0
	src 144.73.20.30 dst 87.172.164.149
			proto esp spi 0x1a9609af reqid 16405 mode tunnel
			replay-window 32 flag af-unspec
			auth-trunc hmac(sha1) 0xdb23eb1368f490b286a3e6920a6039d7c0b27ecc 96
			enc cbc(aes) 0x46fcbd56399a75236c23a277b4cf98bf
			encap type espinudp sport 4500 dport 63453 addr 0.0.0.0

**$ ip xfrm policy**

	src 192.168.2.0/24 dst 192.168.1.0/24
			dir out priority 2344
			tmpl src 0.0.0.0 dst 0.0.0.0
					proto esp reqid 0 mode transport
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket out priority 0
	src 0.0.0.0/0 dst 0.0.0.0/0
			socket in priority 0

Thank you in Advance for any ideas!

**Comment by letoams** [comment]
On Mon, 30 Jan 2017, Bubelbub wrote:

> conn Vereinsnetz
>    left=srv.dyndns.org
>    leftsubnet=192.168.2.0/255.255.255.0
>    right=stick.dyndns.org
>    rightsubnet=192.168.1.0/255.255.255.0

> 
> But if I try...
> 
> $ ping 192.168.2.1 or $ ping 192.168.1.1 (from Client -> Server or Server -> Client) I get no response.
> TCPDump shows nothing.
> The ping said No route to host.
> If I try traceroute then they go a wrong way. (Use internet gateway instead using IPSec tunnel)

Add leftsourceip=192.168.2.1 and rightsourceip=192.168.1.1

> And... If I'm adding the routes manually.
> 
> Server: route add -net 192.168.1.0/24 gw 192.168.2.1 lan-1
> Client: route add -net 192.168.2.0/24 gw 192.168.1.1 lan-1
> 
> Then it work's perfect.

The sourceip= options add those routes.

Paul

> And in my first sentence I said that somebody said IPSec should work without routes.
> So why it work if I add the route and not if I not add the route!? 👎

The fault here is really the "ping" command.

ping uses the "nearest" IP as source address, and the "nearest" IP
to the remote subnet is the public ip, not the internal lan IP.

If you would have used: ping -I 192.168.1.1 192.168.2.1 it would have
worked without the routes.

Paul
Libreswan Developer

**Comment by Bubelbub** [comment]
WoW, Thanks @letoams / Paul for your reply.
It makes absolutly sense and I learned something about this behavior 👍 

But adding `leftsourceip` and `rightsourceip` didn't worked for me.
No route was added to the routing table.
And the ping not work, too.

**BUT** your `ping -I 192.168.1.1 192.168.2.1` command works fine! Without routing entries!
And I found out that if I do something from a third-machine then it works fine, too. 👍 
So only the router has a problem with correct `ping`.

It is really interesting! I don't know why I never tested it from another system except the router 👎 

I will test something different. But your comment was very helpful! 😄 
Thank you very much!
